backend/test/modules/tournament-tree/core/domain/TournamentTree.spec.ts

```diff
@@ -1,118 +1,349 @@
-    const matchEntityIdGen = NumberIdGeneratorStub(1000, 'match');
-    const tournamentTeamsList = generateTournamentTeamsList(teamEntityIdGen, 4);
-
-    // //When
-    const tournamentTree: TournamentTree = TournamentTree.createSingleTournamentTree({
-      tournamentId: testTournamentId.generate(),
-      tournamentTeams: tournamentTeamsList,
-      entityIdGenerator: matchEntityIdGen,
+  describe('Single Tournament Tree', () => {
+    const testTournamentId = NumberIdGeneratorStub(1000, 'testTournamentId');
+
+    it('CreateTournamentTree | Create correct 4 teams tree', async () => {
+      //Given
+      const teamEntityIdGen = NumberIdGeneratorStub(1000, 'team');
+      const matchEntityIdGen = NumberIdGeneratorStub(1000, 'match');
+      const tournamentTeamsList = generateTournamentTeamsList(teamEntityIdGen, 4);
+
+      // //When
+      const tournamentTree: TournamentTree = TournamentTree.createSingleTournamentTree({
+        tournamentId: testTournamentId.generate(),
+        tournamentTeams: tournamentTeamsList,
+        entityIdGenerator: matchEntityIdGen,
+      });
+
+      const expectedWinnerTree: FightingTeamsGroup[] = [
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_3'),
+          firstTeam: tournamentTeamsList[0],
+          secondTeam: tournamentTeamsList[3],
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_1'),
+          matchNumberInSequence: 1,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_2'),
+          firstTeam: tournamentTeamsList[2],
+          secondTeam: tournamentTeamsList[1],
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_1'),
+          matchNumberInSequence: 2,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_1'),
+          firstTeam: undefined,
+          secondTeam: undefined,
+          fightingTeamsGroupLevel: 1,
+          nextMatchId: undefined,
+          matchNumberInSequence: 3,
+        },
+      ].map((item) => FightingTeamsGroup.fromObj(item));
+
+      // Then
+      expect(tournamentTree.getTournamentTreeArray()).toIncludeSameMembers(expectedWinnerTree);
+    });
+
+    it('CreateTournamentTree | Create correct 6 teams tree', async () => {
+      //Given
+      const teamEntityIdGen = NumberIdGeneratorStub(1000, 'team');
+      const matchEntityIdGen = NumberIdGeneratorStub(1000, 'match');
+      const tournamentTeamsList = generateTournamentTeamsList(teamEntityIdGen, 6);
+
+      // //When
+      const tournamentTree: TournamentTree = TournamentTree.createSingleTournamentTree({
+        tournamentId: testTournamentId.generate(),
+        tournamentTeams: tournamentTeamsList,
+        entityIdGenerator: matchEntityIdGen,
+      });
+
+      const expectedWinnerTree: FightingTeamsGroup[] = [
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_7'),
```

Comment:
Nie rozumiem trochę tego property w meczach. Dlaczego to jest potrzebne? 
Teraz nie wystarczyłby nam matchNumberInSequence?

Reply:
Tak, wczoraj do tego doszedłem.
Choć z tego co pamiętam to podczas procesu tworzenia drzewa ID jest niezbędne, ale mogłoby być później usuwane.
Zmienną matchNumberInSequence dodałem teraz, na id opierałem się wcześniej.

Jeśli chodzi o obiekt "FightingTeamsGroupId" to uważam, że tworzenie go było błędem na samym początko. Opierałem sie na innym komponencie i niepotrzebnie to również zaciągnąłem z niego. Był to błąd z mojej strony